libpromises/eval_context.c

```diff
@@ -1443,6 +1450,8 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         return false;
     }
 
+    StringSetAdd(ctx->all_classes, SafeStringDuplicate(name));
```

Comment:
There is already a number of classes being set before even the triggering setting is checked for being enabled which makes it a bit more complicated to reliably switch on-off without being sure we log all classes that get set.